test/pummel/test-http-upload-timeout.js

```diff
@@ -57,7 +60,8 @@ server.listen(0, function() {
 
       function ping() {
         const nextPing = (Math.random() * 900).toFixed();
-        if (nextPing > 600) {
+        // Do not set too big to prevent req.socket from emitting timeout
+        if (nextPing > 300) {
```

Comment:
Doesn't this mean that, with 2/3 probability, not a single `'ping'` will be sent to the server? And the chance of sending more than one `'ping'` is only about 10 %. Isn't sending `'ping'` important for the purpose of the test?

Reply:
Yes, you are right. Maybe i can refactor the code like this.
```
 for (let i = 0; i < 10; i++) {
    connections++;
    let count = 0;
    setTimeout(function() {
      const request = http.request({
        port: server.address().port,
        method: 'POST',
        path: '/'
      });

      function ping() {
        if (++count === 10) {
          request.end();
          return;
        }
        const nextPing = (Math.random() * 900).toFixed();
        request.write('ping');
        setTimeout(ping, nextPing);
      }
      ping();
    }, i * 50);
  }
```